management/operations.go

```diff
@@ -18,14 +18,17 @@ type operation struct {
 	Error          AzureError
 }
 
+// RequestId is assigned by Azure API and can be used to look up the status of an operation
+type RequestId string
+
 //getOperationStatus gets an operation given the operation ID.
-func (client *Client) getOperationStatus(operationId string) (*operation, error) {
-	if operationId == "" {
-		return nil, fmt.Errorf(errParamNotSpecified, "operationId")
+func (client *Client) getOperationStatus(requestId RequestId) (*operation, error) {
```

Comment:
beyond scope of this PR but can be fixed here,too: this shouldn't return pointer.